scene/resources/tile_set.cpp

```diff
@@ -5851,7 +5851,7 @@ void TileData::_get_property_list(List<PropertyInfo> *p_list) const {
 			Variant default_val;
 			Callable::CallError error;
 			Variant::construct(custom_data[i].get_type(), default_val, nullptr, 0, error);
-			property_info = PropertyInfo(tile_set->get_custom_data_layer_type(i), vformat("custom_data_%d", i), PROPERTY_HINT_NONE, String(), PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_NIL_IS_VARIANT);
+			property_info = PropertyInfo(tile_set->get_custom_data_layer_type(i), vformat("custom_data_%d%s%s", i, tile_set->get_custom_data_layer_name(i).is_empty() ? "" : ": ", tile_set->get_custom_data_layer_name(i)), PROPERTY_HINT_NONE, String(), PROPERTY_USAGE_DEFAULT | PROPERTY_USAGE_NIL_IS_VARIANT);
```

Comment:
The condition is unnecessary. If name is empty, it will display `""` the same.

Reply:
Well, the idea behind this is to keep the existing behavior unchanged for nameless fields. That's exactly what's happening in the fourth field in the second screenshot. 

In other words: the empty check determines whether to insert a colon separator between de field index and name.

Otherwise, the index would be followed by a colon with nothing on the right hand.

Reply:
Ah right, I didn't notice the color. Sorry.

Reply:
No problem!